subsys/bluetooth/host/hci_core.c

```diff
@@ -5939,13 +5932,31 @@ int bt_le_adv_start_internal(const struct bt_le_adv_param *param,
 		if (err) {
 			return err;
 		}
+
+		if (param->options & BT_LE_ADV_OPT_CONNECTABLE) {
+			conn = bt_conn_add_le(param->id, BT_ADDR_LE_NONE);
```

Comment:
Since you set the address to `BT_ADDR_LE_NONE` here, don't you need to update code somewhere else to set the real address once we have a connection?

Reply:
The same code that handled writing the addresses when the connection object was created in the connection complete event will overwrite the addresses in the connection complete event (same for all types of connections).